core/state_processor_test.go

```diff
@@ -354,3 +359,51 @@ func GenerateBadBlock(parent *types.Block, engine consensus.Engine, txs types.Tr
 	// Assemble and return the final block for sealing
 	return types.NewBlock(header, txs, nil, receipts, trie.NewStackTrie(nil), nil, true)
 }
+
+func TestStateProcessor_Process(t *testing.T) {
+	database := rawdb.NewMemoryDatabase()
+	gasLimit := uint64(math.MaxUint64)
+
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	auth, err := bind.NewKeyedTransactorWithChainID(key, params.TestingChainID)
+	require.NoError(t, err)
+
+	balance := big.NewInt(0).SetUint64(math.MaxUint64)
+	alloc := make(GenesisAlloc)
+	alloc[auth.From] = GenesisAccount{
+		Balance: balance,
+	}
+
+	cpcfg := params.TestChainConfig
+	cpcfg.ChainID = params.TestingChainID
+	genesis := Genesis{Config: cpcfg, GasLimit: gasLimit, Alloc: alloc, Coinbase: common.HexToAddress("0x0100000000000000000000000000000000000000")}
+	b := genesis.MustCommit(database)
```

Comment:
this line should be moved below IMO, since it's needed for 386 and not related to the lines directly above and below it

Reply:
Block needs to be committed to the database, because database is used in creating blockchain